src/components/nabla/Pools/Swap/WithdrawLiquidity/useWithdrawLiquidity.ts

```diff
@@ -1,70 +1,129 @@
+import { useErc20ContractBalance } from '../../../../../hooks/nabla/useErc20ContractBalance';
+import { useContractWrite } from '../../../../../hooks/nabla/useContractWrite';
+import { useQueryClient } from '@tanstack/react-query';
+import { cacheKeys } from '../../../../../constants/cache';
+import { useQuote } from '../../../../../hooks/nabla/useQuote';
+import { MessageCallErrorResult } from '../../../../../hooks/nabla/useContractRead';
+
+interface WithdrawLiquidityValues {
+  amount: string;
+}
 
-export const useWithdrawLiquidity = (poolAddress: string, tokenAddress: string) => {
+const schema = Yup.object<WithdrawLiquidityValues>().shape({
+  amount: Yup.string().required(),
+});
+
+function parseQuoteError(error: MessageCallErrorResult): string {
+  switch (error.type) {
+    case 'error':
+      return 'Cannot determine value of shares';
+    case 'panic':
+      return error.errorCode === 0x11
+        ? 'The input amount is too large. You cannot directly redeem all LP tokens right now. Try to redeem from the backstop pool instead.'
+        : 'Cannot determine value of shares';
+    case 'reverted':
+      return 'Cannot determine value of shares';
+  }
+}
+
+export const useSwapPoolWithdrawLiquidity = (
+  swapPoolAddress: string,
+  tokenAddress: string,
+  poolTokenDecimals: number,
+  lpTokenDecimals: number,
+) => {
   const queryClient = useQueryClient();
-  const { indexerUrl } = useGetAppDataByTenant('nabla').data || {};
   const toggle = useModalToggle();
 
-  const balanceQuery = useContractBalance({ contractAddress: tokenAddress, decimals: FixedU128Decimals });
-  const depositQuery = useContractBalance({ contractAddress: poolAddress, decimals: FixedU128Decimals });
+  const balanceQuery = useErc20ContractBalance(erc20WrapperAbi, {
+    contractAddress: tokenAddress,
+    decimals: poolTokenDecimals,
+  });
+
+  const depositQuery = useErc20ContractBalance(swapPoolAbi, {
+    contractAddress: swapPoolAddress,
+    decimals: lpTokenDecimals,
+  });
 
   const form = useForm<WithdrawLiquidityValues>({
     resolver: yupResolver(schema),
     defaultValues: {
-      amount: 0,
+      amount: undefined,
     },
   });
-  const { handleSubmit, reset } = form;
 
   const mutation = useContractWrite({
     abi: swapPoolAbi,
-    address: poolAddress,
+    address: swapPoolAddress,
     method: 'withdraw',
-    onError: () => {
-      // ? log error - alert not needed as the transaction modal displays the error
-    },
-    onSuccess: () => {
-      reset();
-      balanceQuery.refetch();
-      depositQuery.refetch();
-      queryClient.refetchQueries([cacheKeys.swapPools, indexerUrl]);
+    mutateOptions: {
+      onError: () => {
+        // ? log error - alert not needed as the transaction modal displays the error
+      },
+      onSuccess: () => {
+        form.reset();
+        balanceQuery.refetch();
+        depositQuery.refetch();
+        setTimeout(() => {
```

Comment:
Could we apply the same fix about not using setTimeout here as well? It would help us maintain consistency and avoid potential issues across the application.